app/src/forms/submission/service.js

```diff
@@ -38,60 +35,15 @@ const service = {
     });
   },
 
-  read: async (formSubmissionId, currentUser, permissions = [Permissions.SUBMISSION_READ]) => {
-    const result = await service._fetchSubmissionData(formSubmissionId);
-
-    const checkFormSubmissionsPermission = async () => {
-      if (currentUser.public) return false;
-      return UserFormAccess.query()
-        .modify('filterFormId', result.form.id)
-        .modify('filterUserId', currentUser.id)
-        .modify('filterByAccess', null, null, permissions)
-        .first();
-    };
-
-    const checkSubmissionPermission = async () => {
-      if (currentUser.public) return false;
-      return FormSubmissionUserPermissions.query()
-        .modify('filterSubmissionId', formSubmissionId)
-        .modify('filterUserId', currentUser.id)
-        .modify('filterByPermissions', permissions)
-        .first();
-    };
-
-    const isDeleted = result.submission.deleted;
-    const isDraft = result.submission.draft;
-    const publicAllowed = result.form.identityProviders.find(p => p.code === 'public') !== undefined;
-    const idpAllowed = result.form.identityProviders.find(p => p.code === currentUser.idp) !== undefined;
-
-    // check against the public and user's identity provider permissions...
-    if (!isDraft && !isDeleted) {
-      if (publicAllowed || idpAllowed) return result;
-    }
-
-    // check against the form level permissions assigned to the user...
-    const formSubmissionsPermission = await checkFormSubmissionsPermission();
-    if (!isDeleted && formSubmissionsPermission) return result;
-
-    // check against the submission level permissions assigned to the user...
-    const submissionPermission = await checkSubmissionPermission();
-    if (submissionPermission) return result;
-
-    // no access to this submission...
-
-    throw new Problem(401, 'You do not have access to this submission.');
+  read: async (formSubmissionId) => {
+    return await service._fetchSubmissionData(formSubmissionId);
```

Comment:
If we're expecting a promise from _fetchSubmissionData and read function is already async, we could make this function a synchronous passthrough instead and avoid unnecessary async/await tackons:
```js
  read: (formSubmissionId) => service._fetchSubmissionData(formSubmissionId),
```